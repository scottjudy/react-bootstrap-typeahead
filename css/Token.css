/* Token */
.bootstrap-tokenizer .token {
  background-color: #e7f4ff;
  border: 0;
  border-radius: 2px;
  color: #1f8dd6;
  display: inline-block;
  line-height: 1em;
  padding: 4px 7px;
  position: relative;
}
.bootstrap-tokenizer .token:focus,
.bootstrap-tokenizer .token-removeable {
  padding-right: 21px;
}
.bootstrap-tokenizer .token-selected {
  background-color: #1f8dd6;
  color: #fff;
  outline: none;
  text-decoration: none;
}

.bootstrap-tokenizer .token {
  margin: 0 3px 3px 0;
}

.bootstrap-tokenizer .token .close-button {
  bottom: 0;
  padding: 3px 7px;
  position: absolute;
  right: 0;
  top: 0;
}
