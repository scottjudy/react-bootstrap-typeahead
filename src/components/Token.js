// @flow

import cx from 'classnames';
import React, { forwardRef } from 'react';

import type { Node } from 'react';

import ClearButton from './ClearButton';

import { withToken } from '../behaviors/token';
import { isFunction } from '../utils';

import type { OptionHandler } from '../types';

type Props = {
  active: boolean,
  children?: Node,
  className?: string,
  disabled?: boolean,
  href?: string,
  onRemove?: OptionHandler,
  readOnly?: boolean,
  tabIndex?: number,
};

const InteractiveToken = forwardRef<Props, ?HTMLElement>(
  ({ active, children, className, onRemove, tabIndex, ...props }, ref) => (
    <div
      {...props}
      className={cx(
        'rbt-token',
        'rbt-token-removeable',
        {
          'rbt-token-active': !!active,
        },
        className
      )}
      ref={ref}
      tabIndex={tabIndex || 0}>
      {children}
      <ClearButton
        className="rbt-token-remove-button"
        label="Remove"
        onClick={onRemove}
        tabIndex={-1}
      />
    </div>
  )
);

const StaticToken = ({ children, className, disabled, href }) => {
  const classnames = cx(
    'rbt-token',
    {
      'rbt-token-disabled': disabled,
    },
    className
  );

  if (href && !disabled) {
    return (
      <a className={classnames} href={href}>
        {children}
      </a>
    );
  }

  return <div className={classnames}>{children}</div>;
};

/**
 * Token
 *
 * Individual token component, generally displayed within the TokenizerInput
 * component, but can also be rendered on its own.
 */
const Token = forwardRef<Props, ?HTMLElement>((props, ref) => {
  const { disabled, onRemove, readOnly } = props;

  return !disabled && !readOnly && isFunction(onRemove) ? (
    <InteractiveToken {...props} ref={ref} />
  ) : (
    <StaticToken {...props} />
  );
});

export default withToken(Token);
