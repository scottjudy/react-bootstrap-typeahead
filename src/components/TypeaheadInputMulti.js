// @flow

/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */

import cx from 'classnames';
import React, { type Node } from 'react';

import Hint, { type ShouldSelect } from './Hint';
import Input from './Input';

import { isSelectable } from '../utils';
import withClassNames from '../behaviors/classNames';

import { BACKSPACE } from '../constants';

import type { InputProps, Option, RefCallback, ReferenceElement } from '../types';

type Props = {
  ...InputProps,
  children: Node,
  referenceElementRef: RefCallback<ReferenceElement>,
  selected: Option[],
  shouldSelectHint?: ShouldSelect,
};

class TypeaheadInputMulti extends React.Component<Props> {
  wrapperRef = React.createRef<HTMLDivElement>();
  _input: ?HTMLInputElement;

  render() {
    const {
      children,
      className,
      inputClassName,
      inputRef,
      placeholder,
      referenceElementRef,
      selected,
      shouldSelectHint,
      ...props
    } = this.props;

    return (
      <div
        className={cx('rbt-input-multi', className)}
        disabled={props.disabled}
        onClick={this._handleContainerClickOrFocus}
        onFocus={this._handleContainerClickOrFocus}
        ref={referenceElementRef}
        tabIndex={-1}>
        <div className="rbt-input-wrapper" ref={this.wrapperRef}>
          {children}
          <Hint shouldSelect={shouldSelectHint}>
            <Input
              {...props}
              className={inputClassName}
              onKeyDown={this._handleKeyDown}
              placeholder={selected.length ? '' : placeholder}
              ref={this.getInputRef}
              style={{
                backgroundColor: 'transparent',
                border: 0,
                boxShadow: 'none',
                cursor: 'inherit',
                outline: 'none',
                padding: 0,
                width: '100%',
                zIndex: 1,
              }}
            />
          </Hint>
        </div>
      </div>
    );
  }

  getInputRef = (input: ?HTMLInputElement) => {
    this._input = input;
    this.props.inputRef(input);
  }

  /**
   * Forward click or focus events on the container element to the input.
   */
  _handleContainerClickOrFocus = (e: SyntheticEvent<HTMLElement>) => {
    // Don't focus the input if it's disabled.
    if (this.props.disabled) {
      e.currentTarget.blur();
      return;
    }

    // Move cursor to the end if the user clicks outside the actual input.
    const inputNode = this._input;
    if (!inputNode) {
      return;
    }

    if (e.currentTarget !== inputNode && isSelectable(inputNode)) {
      inputNode.selectionStart = inputNode.value.length;
    }

    inputNode.focus();
  }

  _handleKeyDown = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
    const { onKeyDown, selected, value } = this.props;

    switch (e.keyCode) {
      case BACKSPACE:
        if (e.currentTarget === this._input && selected.length && !value) {
          // Prevent browser from going back.
          e.preventDefault();

          // If the input is selected and there is no text, focus the last
          // token when the user hits backspace.
          if (this.wrapperRef.current) {
            const { children } = this.wrapperRef.current;
            const lastToken = children[children.length - 2];
            lastToken && lastToken.focus();
          }
        }
        break;
      default:
        break;
    }

    onKeyDown(e);
  }
}

export default withClassNames(TypeaheadInputMulti);
