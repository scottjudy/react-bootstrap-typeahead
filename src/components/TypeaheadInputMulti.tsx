/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */

import cx from 'classnames';
import React, { FocusEvent, KeyboardEvent, MouseEvent, ReactNode } from 'react';

import Hint from './Hint';
import Input from './Input';

import { isSelectable } from '../utils';
import withClassNames from '../behaviors/classNames';

import { Option, RefElement, TypeaheadInputProps } from '../types';

interface Props extends TypeaheadInputProps {
  children: ReactNode;
  selected: Option[];
}

class TypeaheadInputMulti extends React.Component<Props> {
  wrapperRef = React.createRef<HTMLDivElement>();
  _input: RefElement<HTMLInputElement> = null;

  render() {
    const {
      children,
      className,
      inputClassName,
      inputRef,
      placeholder,
      referenceElementRef,
      selected,
      shouldSelectHint,
      ...props
    } = this.props;

    return (
      <div
        className={cx('rbt-input-multi', className)}
        onClick={this._handleContainerClickOrFocus}
        onFocus={this._handleContainerClickOrFocus}
        ref={referenceElementRef}
        tabIndex={-1}>
        <div className="rbt-input-wrapper" ref={this.wrapperRef}>
          {children}
          <Hint shouldSelect={shouldSelectHint}>
            <Input
              {...props}
              className={inputClassName}
              onClick={this._handleClick}
              onKeyDown={this._handleKeyDown}
              placeholder={selected.length ? '' : placeholder}
              ref={this.getInputRef}
              style={{
                backgroundColor: 'transparent',
                border: 0,
                boxShadow: 'none',
                cursor: 'inherit',
                outline: 'none',
                padding: 0,
                width: '100%',
                zIndex: 1,
              }}
            />
          </Hint>
        </div>
      </div>
    );
  }

  getInputRef = (input: RefElement<HTMLInputElement>) => {
    this._input = input;
    this.props.inputRef(input);
  };

  _handleClick = (e: MouseEvent<HTMLInputElement>) => {
    // Prevent clicks on the input from bubbling up to the container,
    // which then re-focuses the input.
    e.stopPropagation();
    this.props.onClick && this.props.onClick(e);
  };

  /**
   * Forward click or focus events on the container element to the input.
   */
  _handleContainerClickOrFocus = (
    e: MouseEvent<HTMLElement> | FocusEvent<HTMLElement>
  ) => {
    // Don't focus the input if it's disabled.
    if (this.props.disabled) {
      e.currentTarget.blur();
      return;
    }

    // Move cursor to the end if the user clicks outside the actual input.
    const inputNode = this._input;

    if (!inputNode) {
      return;
    }

    if (isSelectable(inputNode)) {
      inputNode.selectionStart = inputNode.value.length;
    }

    inputNode.focus();
  };

  _handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    const { onKeyDown, selected, value } = this.props;

    switch (e.key) {
      case 'Backspace':
        if (e.currentTarget === this._input && selected.length && !value) {
          // Prevent browser from going back.
          e.preventDefault();

          // If the input is selected and there is no text, focus the last
          // token when the user hits backspace.
          if (this.wrapperRef.current) {
            const { children } = this.wrapperRef.current;
            const lastToken = children[children.length - 2];
            lastToken && (lastToken as HTMLElement).focus();
          }
        }
        break;
      default:
        break;
    }

    onKeyDown && onKeyDown(e);
  };
}

export default withClassNames(TypeaheadInputMulti);
