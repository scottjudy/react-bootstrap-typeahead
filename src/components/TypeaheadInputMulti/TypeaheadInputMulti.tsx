/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */

import cx from 'classnames';
import React, { FocusEvent, KeyboardEvent, MouseEvent, ReactNode } from 'react';

import Hint from '../Hint';
import Input from '../Input';

import { isSelectable } from '../../utils';
import withClassNames from '../../behaviors/classNames';

import { Option, RefElement, TypeaheadInputProps } from '../../types';

export interface TypeaheadInputMultiProps extends TypeaheadInputProps {
  children: ReactNode;
  selected: Option[];
}

class TypeaheadInputMulti extends React.Component<TypeaheadInputMultiProps> {
  wrapperRef = React.createRef<HTMLDivElement>();
  _input: RefElement<HTMLInputElement> = null;

  render() {
    const {
      children,
      className,
      inputClassName,
      inputRef,
      placeholder,
      referenceElementRef,
      selected,
      ...props
    } = this.props;

    return (
      <div
        className={cx('rbt-input-multi', className)}
        onClick={this._handleContainerClickOrFocus}
        onFocus={this._handleContainerClickOrFocus}
        ref={referenceElementRef}
        tabIndex={-1}>
        <div className="rbt-input-wrapper" ref={this.wrapperRef}>
          {children}
          <Hint>
            <Input
              {...props}
              className={inputClassName}
              onKeyDown={this._handleKeyDown}
              placeholder={selected.length ? '' : placeholder}
              ref={this.getInputRef}
              style={{
                backgroundColor: 'transparent',
                border: 0,
                boxShadow: 'none',
                cursor: 'inherit',
                outline: 'none',
                padding: 0,
                width: '100%',
                zIndex: 1,
              }}
            />
          </Hint>
        </div>
      </div>
    );
  }

  getInputRef = (input: RefElement<HTMLInputElement>) => {
    this._input = input;
    this.props.inputRef(input);
  };

  /**
   * Forward click or focus events on the container element to the input.
   */
  _handleContainerClickOrFocus = (
    e: MouseEvent<HTMLElement> | FocusEvent<HTMLElement>
  ) => {
    // Don't focus the input if it's disabled.
    if (this.props.disabled) {
      e.currentTarget.blur();
      return;
    }

    const inputNode = this._input;
    if (
      !inputNode ||
      // Ignore if the clicked element is a child of the container, ie: a token
      // or the input itself.
      (e.currentTarget.contains(e.target as HTMLElement) &&
        e.currentTarget !== e.target)
    ) {
      return;
    }

    if (isSelectable(inputNode)) {
      // Move cursor to the end if the user clicks outside the actual input.
      inputNode.selectionStart = inputNode.value.length;
    }

    inputNode.focus();
  };

  _handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    const { onKeyDown, selected, value } = this.props;

    if (e.key === 'Backspace' && selected.length && !value) {
      // Prevent browser from going back.
      e.preventDefault();

      // If the input is selected and there is no text, focus the last
      // token when the user hits backspace.
      if (this.wrapperRef.current) {
        const { children } = this.wrapperRef.current;
        const lastToken = children[children.length - 2];
        lastToken && (lastToken as HTMLElement).focus();
      }
    }

    onKeyDown && onKeyDown(e);
  };
}

export default withClassNames(TypeaheadInputMulti);
