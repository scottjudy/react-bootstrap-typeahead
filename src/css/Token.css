/* Token */
.token {
  background-color: #e7f4ff;
  border: 0;
  border-radius: 2px;
  color: #1f8dd6;
  float: left;
  line-height: 1em;
  margin: 0 3px 3px 0;
  padding: 4px 21px 4px 7px;
  position: relative;
}
.token:focus,
.token-selected {
  background-color: #1f8dd6;
  color: #fff;
  outline: none;
  text-decoration: none;
}

.token-close-button {
  bottom: 0;
  padding: 3px 7px;
  position: absolute;
  right: 0;
  top: 0;
}
