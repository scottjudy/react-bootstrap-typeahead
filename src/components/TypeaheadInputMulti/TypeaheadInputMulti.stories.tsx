/* eslint-disable sort-keys,import/no-extraneous-dependencies */

import React, { ChangeEvent, useState } from 'react';
import { Story, Meta } from '@storybook/react';

import Token from '../Token';
import TypeaheadInputMulti, {
  TypeaheadInputMultiProps,
} from './TypeaheadInputMulti';

import options from '../../tests/data';
import { HintProvider, noop } from '../../tests/helpers';

export default {
  title: 'Components/TypeaheadInputMulti',
  component: TypeaheadInputMulti,
} as Meta;

const selected = options.slice(1, 4);

const defaultProps = {
  children: selected.map((option) => (
    <Token key={option.name} option={option} onRemove={noop}>
      {option.name}
    </Token>
  )),
  selected,
};

interface Args extends TypeaheadInputMultiProps {
  hintText?: string;
}

const Template: Story<Args> = ({ hintText = '', ...args }) => {
  const [value, setValue] = useState(args.value);
  const [inputNode, setInputNode] = useState<HTMLInputElement | null>(null);

  return (
    <HintProvider hintText={hintText} inputNode={inputNode}>
      <TypeaheadInputMulti
        {...args}
        inputRef={setInputNode}
        onChange={(e: ChangeEvent<HTMLInputElement>) =>
          setValue(e.target.value)
        }
        referenceElementRef={noop}
        value={value}
      />
    </HintProvider>
  );
};

export const Default = Template.bind({});
Default.args = {
  ...defaultProps,
};

export const WithHint = Template.bind({});
WithHint.args = {
  ...defaultProps,
  hintText: 'california',
  value: 'cali',
};

export const Disabled = Template.bind({});
Disabled.args = {
  ...defaultProps,
  children: selected.map((option) => (
    <Token disabled key={option.name} option={option} onRemove={noop}>
      {option.name}
    </Token>
  )),
  disabled: true,
};
