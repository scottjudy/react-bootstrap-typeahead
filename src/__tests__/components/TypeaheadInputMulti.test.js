import React from 'react';

import Token from '../../components/Token';
import TypeaheadInputMulti from '../../components/TypeaheadInputMulti';

import options from '../data';

import {
  fireEvent,
  getHint,
  getInput,
  getTokens,
  noop,
  prepareSnapshot,
  render,
  screen,
  TestProvider,
  userEvent,
} from '../helpers';

const TestComponent = ({ context, props }) => {
  const selected = options.slice(1, 4);

  return (
    <TestProvider {...context} multiple onKeyDown={noop} selected={selected}>
      {({ getInputProps }) => (
        <TypeaheadInputMulti
          {...getInputProps()}
          {...props}
          selected={selected}>
          {selected.map((option) => (
            <Token key={option.name} option={option} onRemove={noop}>
              {option.name}
            </Token>
          ))}
        </TypeaheadInputMulti>
      )}
    </TestProvider>
  );
};

describe('<TypeaheadInputMulti>', () => {
  it('renders a snapshot', () => {
    expect(prepareSnapshot(<TestComponent />)).toMatchSnapshot();
  });

  it('displays the selected text', () => {
    const text = 'foo';
    render(<TestComponent context={{ text }} />);
    expect(getInput(screen).value).toBe(text);
  });

  it('renders a multi-select input with tokens', () => {
    const { container } = render(<TestComponent />);
    // Find the token close buttons as a proxy for the token itself.
    expect(getTokens(container)).toHaveLength(3);
  });

  it('displays a hint and calls `shouldSelectHint`', () => {
    const initialItem = options[0];
    const shouldSelectHint = jest.fn();

    const { container } = render(
      <TestComponent
        context={{
          initialItem,
          isFocused: true,
          isMenuShown: true,
          text: 'Al',
        }}
        props={{
          shouldSelectHint,
        }}
      />
    );

    const hintInput = getHint(container);
    expect(hintInput).toHaveValue(initialItem.name);

    // No need to test the logic for `shouldSelectHint` here; just make sure
    // it's passed through to the `Hint` component and called.
    getInput(screen).focus();
    userEvent.tab();
    expect(shouldSelectHint).toHaveBeenCalledTimes(1);
  });

  it('does not focus a disabled input', () => {
    render(
      <React.Fragment>
        <TestComponent props={{ 'data-testid': 'enabled-input' }} />
        <TestComponent
          props={{ 'data-testid': 'disabled-input', disabled: true }}
        />
      </React.Fragment>
    );

    const enabledInput = screen.getByTestId('enabled-input');
    const disabledInput = screen.getByTestId('disabled-input');

    expect(enabledInput).not.toBeDisabled();
    expect(disabledInput).toBeDisabled();

    userEvent.click(enabledInput);
    expect(enabledInput).toHaveFocus();

    userEvent.click(disabledInput);
    expect(disabledInput).not.toHaveFocus();
  });

  it('prevents clicks on the input from bubbling', () => {
    const onClick = jest.fn();
    render(<TestComponent context={{ text: 'calif' }} props={{ onClick }} />);

    const input = getInput(screen);
    input.selectionStart = 2;
    // userEvent.click triggers the wrong behavior for some reason.
    fireEvent.click(input);

    // Cursor shouldn't move when the input is clicked once.
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(input.selectionStart).toBe(2);
  });

  it('calls the keydown handler', () => {
    const onKeyDown = jest.fn();
    render(<TestComponent props={{ onKeyDown }} />);

    getInput(screen).focus();
    userEvent.keyboard('{enter}');

    expect(onKeyDown).toHaveBeenCalledTimes(1);
  });

  it('focuses the last token', () => {
    const { container } = render(<TestComponent />);

    getInput(screen).focus();
    userEvent.keyboard('{backspace}');

    const tokens = getTokens(container);
    const lastToken = tokens[tokens.length - 1];

    expect(lastToken).toHaveFocus();
  });

  it('does not focus the last token when the input has a value', () => {
    render(<TestComponent context={{ text: 'foo' }} />);

    const input = getInput(screen);
    input.focus();
    userEvent.keyboard('{backspace}');

    expect(input).toHaveFocus();
  });
});
