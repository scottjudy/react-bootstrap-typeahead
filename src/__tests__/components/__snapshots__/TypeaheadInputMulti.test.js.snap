// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeaheadInputMulti> renders a snapshot 1`] = `
<div
  className="rbt-input-multi form-control rbt-input"
  onClick={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <div
    className="rbt-input-wrapper"
  >
    <div
      className="rbt-token rbt-token-removeable"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      Alaska
      <button
        aria-label="Remove"
        className="close rbt-close rbt-token-remove-button"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only visually-hidden"
        >
          Remove
        </span>
      </button>
    </div>
    <div
      className="rbt-token rbt-token-removeable"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      Arizona
      <button
        aria-label="Remove"
        className="close rbt-close rbt-token-remove-button"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only visually-hidden"
        >
          Remove
        </span>
      </button>
    </div>
    <div
      className="rbt-token rbt-token-removeable"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      Arkansas
      <button
        aria-label="Remove"
        className="close rbt-close rbt-token-remove-button"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          aria-hidden="true"
        >
          ×
        </span>
        <span
          className="sr-only visually-hidden"
        >
          Remove
        </span>
      </button>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "height": "100%",
          "position": "relative",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-haspopup="listbox"
        autoComplete="off"
        className="rbt-input-main"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        style={
          Object {
            "backgroundColor": "transparent",
            "border": 0,
            "boxShadow": "none",
            "cursor": "inherit",
            "outline": "none",
            "padding": 0,
            "width": "100%",
            "zIndex": 1,
          }
        }
        type="text"
        value=""
      />
      <input
        aria-hidden={true}
        className="rbt-input-hint"
        readOnly={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "boxShadow": "none",
            "color": "rgba(0, 0, 0, 0.35)",
            "left": 0,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
        tabIndex={-1}
        value=""
      />
    </div>
  </div>
</div>
`;
